python/ray/autoscaler/_private/command_runner.py

```diff
@@ -669,7 +669,7 @@ def _docker_expand_user(self, string, any_char=False):
         if user_pos > -1:
             if self.home_dir is None:
                 self.home_dir = self.ssh_command_runner.run(
-                    "docker exec {} env | grep HOME | cut -d'=' -f2".format(
+                    "docker exec {} env | grep ^HOME= | cut -d'=' -f2".format(
```

Comment:
Should we add a fallback option? (Like if $HOME is **not set**)?

Reply:
Not sure about that. if HOME isn't set we should probably fail loudly basically it basically means user's env is broken.

Reply:
many, many commands rely on it

Reply:
Oh! Well then I'm fine with assuming that it is set :)